File: flutter/lib/src/koorugal/podhu_koorugal/elvan_pothu_attai.dart
import 'package:flutter/material.dart';

/// A completely reusable Elvan Card component.
/// It wraps the content in a Material + InkWell to provide the native
/// Android ripple effect upon tap. It also smoothly animates background
/// color changes when `isSelected` changes.
class ElvanPothuAttai extends StatelessWidget {
  const ElvanPothuAttai({
    super.key,
    required this.child,
    this.onTap,
    this.onLongPress,
    this.isSelected = false,
    this.padding = const EdgeInsets.all(16),
    this.borderRadius = 24.0,
  });

  /// The inner content of the card
  final Widget child;

  /// Triggered on tap
  final VoidCallback? onTap;

  /// Triggered on long press
  final VoidCallback? onLongPress;

  /// Whether the card is selected (changes the background color)
  final bool isSelected;

  /// Inner padding for the content
  final EdgeInsetsGeometry padding;

  /// Corner border radius
  final double borderRadius;

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;

    final defaultBg = isDark ? const Color(0xFF111111) : Colors.white;
    final selectedBg = isDark
        ? const Color(0xFF1A1A1A)
        : Colors.black.withValues(alpha: 0.04);
    
    final bgColor = isSelected ? selectedBg : defaultBg;

    return TweenAnimationBuilder<Color?>(
      tween: ColorTween(begin: bgColor, end: bgColor),
      duration: const Duration(milliseconds: 200),
      builder: (context, color, _) {
        return Material(
          color: color,
          borderRadius: BorderRadius.circular(borderRadius),
          clipBehavior: Clip.antiAlias,
          child: InkWell(
            onTap: onTap,
            onLongPress: onLongPress,
            borderRadius: BorderRadius.circular(borderRadius),
            child: Padding(
              padding: padding,
              child: child,
            ),
          ),
        );
      },
    );
  }
}

File: migrate/composeApp/src/commonMain/kotlin/com/elvan/noolachu/ui/components/ElvanPothuAttai.kt
package com.elvan.noolachu.ui.components

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.background
import androidx.compose.foundation.combinedClickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.interaction.collectIsPressedAsState
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.ripple
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.elvan.noolachu.theme.rememberShellColors

/**
 * A completely reusable Elvan Card component.
 * Exact 1:1 Kotlin Compose port of Flutter's ElvanPothuAttai.
 *
 * It wraps the content to provide the native ripple effect upon tap.
 * It also smoothly animates press scale (0.985f) and background color changes
 * when `isSelected` changes.
 */
@OptIn(ExperimentalFoundationApi::class)
@Composable
fun ElvanPothuAttai(
    modifier: Modifier = Modifier,
    onClick: (() -> Unit)? = null,
    onLongClick: (() -> Unit)? = null,
    isSelected: Boolean = false,
    padding: PaddingValues = PaddingValues(16.dp),
    borderRadius: Dp = 24.dp,
    content: @Composable () -> Unit
) {
    val colors = rememberShellColors()
    val isDark = colors.isDark

    val defaultBg = if (isDark) Color(0xFF111111) else Color.White
    val selectedBg = if (isDark) Color(0xFF1A1A1A) else Color.Black.copy(alpha = 0.04f)
    val bgColor = if (isSelected) selectedBg else defaultBg

    val interactionSource = remember { MutableInteractionSource() }
    val isPressed by interactionSource.collectIsPressedAsState()

    val scale by animateFloatAsState(
        targetValue = if (isPressed && (onClick != null || onLongClick != null)) 0.985f else 1.0f,
        animationSpec = tween(
            durationMillis = if (isPressed) 100 else 200
        ),
        label = "pothuAttaiPressScale"
    )

    val shape = RoundedCornerShape(borderRadius)

    Box(
        modifier = modifier
            .fillMaxWidth()
            .graphicsLayer {
                scaleX = scale
                scaleY = scale
            }
            .clip(shape)
            .background(bgColor)
            .then(
                if (onClick != null || onLongClick != null) {
                    Modifier.combinedClickable(
                        interactionSource = interactionSource,
                        indication = ripple(bounded = true, color = colors.ripple),
                        onClick = { onClick?.invoke() },
                        onLongClick = onLongClick
                    )
                } else Modifier
            )
            .padding(padding)
    ) {
        content()
    }
}
